ntb_hw_switchtec.c

```diff
@@ -1388,53 +1388,6 @@ static int switchtec_ntb_init_crosslink(struct switchtec_ntb *sndev)
 		return -EINVAL;
 	}
 
-	addr = (bar_addrs[0] + SWITCHTEC_GAS_NTB_OFFSET +
-		SWITCHTEC_NTB_REG_DBMSG_OFFSET +
-		sizeof(struct ntb_dbmsg_regs) * sndev->peer_partition);
-
-	offset = addr & (LUT_SIZE - 1);
-	addr -= offset;
-
-	rc = config_rsvd_lut_win(sndev, sndev->mmio_self_ctrl, dbmsg_lut_idx,
-				 sndev->peer_partition, addr);
-	if (rc)
-		return rc;
-
-	sndev->mmio_xlink_dbmsg_win = pci_iomap_range(sndev->stdev->pdev, bar,
-						      dbmsg_lut_idx * LUT_SIZE,
-						      LUT_SIZE);
-	if (!sndev->mmio_xlink_dbmsg_win) {
-		rc = -ENOMEM;
-		return rc;
-	}
-
-	sndev->mmio_peer_dbmsg = sndev->mmio_xlink_dbmsg_win + offset;
-	sndev->nr_rsvd_luts++;
-
-	addr = (bar_addrs[0] + SWITCHTEC_GAS_NTB_OFFSET +
-		SWITCHTEC_NTB_REG_CTRL_OFFSET +
-		sizeof(struct ntb_ctrl_regs) * sndev->peer_partition);
-
-	offset = addr & (LUT_SIZE - 1);
-	addr -= offset;
-
-	rc = config_rsvd_lut_win(sndev, sndev->mmio_self_ctrl,
-				 req_id_lut_idx,
-				 sndev->peer_partition, addr);
-	if (rc)
-		return rc;
-
-	sndev->mmio_xlink_ctrl_win = pci_iomap_range(sndev->stdev->pdev, bar,
-						     req_id_lut_idx * LUT_SIZE,
-						     LUT_SIZE);
-	if (!sndev->mmio_xlink_ctrl_win) {
-		rc = -ENOMEM;
-		return rc;
-	}
-
-	sndev->mmio_xlink_peer_ctrl = sndev->mmio_xlink_ctrl_win + offset;
-	sndev->nr_rsvd_luts++;
-
```

Comment:
Does removing this code not break things? A commit shouldn't break things even if a later commit fixes it. 

In any case it would be better to see the removal and the move in one patch.

Reply:
For the reason that it's not just move some LUT init code to another place, but also introduced an new way to initialize them. I thought it would be easier to follow if I split them into two or more commits, otherwise the commit would looks too big to me.

Reply:
If you're going to split it, it would be better to have one commit do the move without functional changes and one make the changes. No way to tell what's changed in the current method.

Reply:
Kept the removal and move in one commit.